File: Iniciante/1048 - Aumento de Salário/Python/1048.py
PERCENTAGE_OF_READJUSTMENT = [0.15, 0.12, 0.10, 0.07, 0.04]


def main():
    salary = float(input())
    readjustment = define_readjustment_percentage(salary)
    salary_increase = salary * readjustment
    new_salary = salary + salary_increase
    print_result(readjustment, salary_increase, new_salary)


def define_readjustment_percentage(salary):
    if salary <= 400.00:
        return PERCENTAGE_OF_READJUSTMENT[0]
    elif salary <= 800.00:
        return PERCENTAGE_OF_READJUSTMENT[1]
    elif salary <= 1200.00:
        return PERCENTAGE_OF_READJUSTMENT[2]
    elif salary <= 2000.00:
        return PERCENTAGE_OF_READJUSTMENT[3]
    else:
        return PERCENTAGE_OF_READJUSTMENT[4]


def print_result(readjustment, salary_increase, new_salary):
    print(f"Novo salario: {new_salary:.2f}")
    print(f"Reajuste ganho: {salary_increase:.2f}")
    print(f"Em percentual: {readjustment * 100:.0f} %")


if __name__ == "__main__":
    main()

File: Iniciante/1048 - Aumento de Salário/TypeScript/1048.ts
import * as fs from "fs";

const PERCENTAGE_OF_READJUSTMENT: number[] = [0.15, 0.12, 0.1, 0.07, 0.04];

function main(): void {
  const input: string = fs.readFileSync("/dev/stdin", "utf8");
  const salary: number = parseFloat(input);
  const readjustment: number = defineReadjustmentPercentage(salary);
  const salaryIncrease: number = salary * readjustment;
  const newSalary: number = salary + salaryIncrease;
  printResult(readjustment, salaryIncrease, newSalary);
}

function defineReadjustmentPercentage(salary: number): number {
  if (salary <= 400.0) {
    return PERCENTAGE_OF_READJUSTMENT[0];
  } else if (salary <= 800.0) {
    return PERCENTAGE_OF_READJUSTMENT[1];
  } else if (salary <= 1200.0) {
    return PERCENTAGE_OF_READJUSTMENT[2];
  } else if (salary <= 2000.0) {
    return PERCENTAGE_OF_READJUSTMENT[3];
  } else {
    return PERCENTAGE_OF_READJUSTMENT[4];
  }
}

function printResult(readjustment: number, salaryIncrease: number, newSalary: number) {
  console.log(`Novo salario: ${newSalary.toFixed(2)}`);
  console.log(`Reajuste ganho: ${salaryIncrease.toFixed(2)}`);
  console.log(`Em percentual: ${(readjustment * 100).toFixed(0)} %`);
}

main();
